docs/source/user_guide/graphql.md

```diff
@@ -0,0 +1,518 @@
+          type
+        }
+        node {
+          id
+          label
+        }
+      }
+    }
+  }
+}
+```
+
+
+```html
+http://localhost:5000/api/v4/nodes/ffe11/contents/attributes
+```
+
+```graphql
+{ Node(uuid: "dee1f869-c45e-40d9-9f9c-f492f4117f13") { attributes } }
+```
+
+
+```html
+http://localhost:5000/api/v4/nodes/ffe11/contents/attributes?attributes_filter=append_text,is_local
+```
+
+```graphql
+{ Node(uuid: "dee1f869-c45e-40d9-9f9c-f492f4117f13") { attributes(filter: ["append_text", "is_local"]) } }
+```
+
+
+```html
+http://localhost:5000/api/v4/nodes/ffe11/contents/comments
+```
+
+```graphql
+{
+  Node(id: 1011) {
+    Comments {
+      count
+      rows {
+        content
+      }
+    }
+  }
+}
+```
+
+Repository based queries are not yet implemented:
+
+```html
+http://localhost:5000/api/v4/nodes/ffe11/repo/list
+```
+```html
+http://localhost:5000/api/v4/nodes/ffe11/repo/contents?filename="aiida.in"
+```
+```html
+http://localhost:5000/api/v4/nodes/fafdsf/download?download_format=xsf
+```
+
+
+
+```html
+http://localhost:5000/api/v4/nodes?mtime>=2019-04-23
+```
+
+```graphql
+{
+  Nodes(filters: "mtime>=2019-04-23") {
+    count
+    rows {
+        uuid
+    }
+  }
+}
+```
+
+### Processes
+
+NOT YET IMPLEMENTED
+
+```html
+http://localhost:5000/api/v4/processes/8b95cd85/report
+http://localhost:5000/api/v4/calcjobs/sffs241j/input_files
+```
+
+### Computers
+
+```
+http://localhost:5000/api/v4/computers?limit=3&offset=2&orderby=id
+```
+
+```graphql
+{
+  Computers {
+    count
+    rows(limit: 3, offset: 3, orderBy: "id") {
+      id
+    }
+}
+```
+
+```html
+http://localhost:5000/api/v4/computers/5d490d77
+```
+
+```graphql
+{
+  Computer(uuid: "5d490d77") {
+    label
+  }
+}
+```
+
+```html
+http://localhost:5000/api/v4/computers/?scheduler_type=in="slurm","pbs"
+```
+
+```graphql
+{
+  Computers(filters: "scheduler_type IN slurm,pbs") {
+    count
+    rows {
+      scheduler_type
+    }
+  }
+}
+```
+
+```html
+http://localhost:5000/api/v4/computers?orderby=+name
+```
+
+```graphql
+{
+  Computers {
+    rows(orderBy: "name") {
+      id
+    }
+  }
+}
+```
+
+```html
+http://localhost:5000/api/v4/computers/page/1?perpage=5
+```
+
+```graphql
+{
+  Computers {
+    rows(limit: 5) {
+      id
+    }
+  }
+}
+```
+
+### Users
+
+```html
+http://localhost:5000/api/v4/users/
+```
+
+```graphql
+{
+  Users {
+    count
+    rows {
+      id
+    }
+  }
+}
+```
+
+```html
+http://localhost:5000/api/v4/users/?first_name=ilike="aii%"
+```
+
+```graphql
+{
+  Users(filters: "first_name iLIKE 'aii%'") {
+    count
+    rows {
+      email
+      first_name
+      last_name
+      institution
+    }
+  }
+}
+```
+
+### Groups
+
+```
+http://localhost:5000/api/v4/groups/?limit=10&orderby=-user_id
+```
+
+```graphql
+{
+  Groups {
+    count
+    rows(limit: 10, orderBy: "user_id", orderAsc: false) {
+      id
+    }
+  }
+}
+```
+
+```html
+http://localhost:5000/api/v4/groups/a6e5b
+```
+
+```graphql
+{
+  Group(id: 1) {
```

Comment:
as noted in this doc though, the current argument does not allow you to pass in partial UUIDs. Have to have a look how this is done in aiida-core, since Postgresql simply throws an error if you try this.
I guess if we definitely want it, you could do something like test its length and turn it into the filter `{"uuid": {"like": "a6eb%"}` if too short, plus check that only one is returned

Reply:
up to you - it's mentioned in the same page that this is not yet implemented, so it's relatively clear what is meant here.
for me, this implementation does not need to be part of this PR

Reply:
done